consensus/core/src/network/mod.rs

```diff
@@ -61,6 +61,8 @@ where
     fn install_service(&self, network_keypair: NetworkKeyPair, service: Arc<S>);
 
     /// Stops the network service.
+    // TODO: Investigate if we need to use async fn here
+    #[allow(async_fn_in_trait)]
```

Comment:
Why is this attribute needed?

Reply:
hhm same question from my side as well

Reply:
I do as the linter tells me 🙇🏽 

This was the warning was a result of changing the visibility for `NetworkManager`. The todo was to see if we should go with the suggestion in the linter or another approach.

```
mod.rs(66, 5): original diagnostic
use of `async fn` in public traits is discouraged as auto trait bounds cannot be specified
you can suppress this lint if you plan to use the trait only in your own code, or do not care about auto traits like `Send` on the `Future`
`#[warn(async_fn_in_trait)]` on by default

mod.rs(66, 5): you can alternatively desugar to a normal `fn` that returns `impl Future` and add any desired bounds such as `Send`, but these cannot be relaxed without a breaking API change: ` -> impl std::future::Future<Output = ()> + Send`
```